retrievalmarket/impl/clientstates/client_fsm.go

```diff
@@ -179,7 +181,10 @@ var ClientEvents = fsm.Events{
 		FromMany(
 			rm.DealStatusOngoing,
 			rm.DealStatusBlocksComplete,
-			rm.DealStatusFundsNeeded).To(rm.DealStatusFundsNeeded).
+			rm.DealStatusFundsNeeded,
+			rm.DealStatusFundsNeededLastPayment).To(rm.DealStatusFundsNeeded).
+		From(rm.DealStatusSendFunds).To(rm.DealStatusOngoing).
```

Comment:
So we are moving to `Ongoing` here because we might be stuck in the `DealStatusSendFunds ` state -> letting the Ongoing state handler check if we have any dues left and if we do -> that handler will trigger a `ClientEventPaymentRequested ` again ? 

That's the idea behind adding this particular transition, right ?